`TestIntegrations/ControlMaster` flakiness

## Failure

#### Link(s) to logs

- https://console.cloud.google.com/cloud-build/builds/bd1b8d88-ca7b-4007-9677-3ba6c77be42c?project=ci-account

#### Relevant snippet

I couldn't find any clear logs detailing the precise failure 🤷‍♂️. This is the second time I've seen this one today, I didn't report it the first time.

```
    --- FAIL: TestIntegrations/ControlMaster (6.98s)
```

Fails locally for me with:
```
--- FAIL: TestIntegrations (0.41s)
    --- FAIL: TestIntegrations/ControlMaster (0.41s)
        integration_test.go:3532: Unexpected error: kex_exchange_identification: Connection closed by remote host
            Connection closed by UNKNOWN port 65535
FAIL
FAIL	github.com/gravitational/teleport/integration	5.859s
FAIL
```

--- a/integration/helpers/helpers.go
+++ b/integration/helpers/helpers.go
@@ -92,8 +92,8 @@ func ExternalSSHCommand(o CommandOptions) (*exec.Cmd, error) {
 	// ControlMaster is often used by applications like Ansible.
 	if o.ControlPath != "" {
 		execArgs = append(execArgs, "-oControlMaster=auto")
-		execArgs = append(execArgs, "-oControlPersist=1s")
-		execArgs = append(execArgs, "-oConnectTimeout=2")
+		execArgs = append(execArgs, "-oControlPersist=0")
+		execArgs = append(execArgs, "-oConnectTimeout=10")
 		execArgs = append(execArgs, fmt.Sprintf("-oControlPath=%v", o.ControlPath))
 	}
 

--- a/integration/integration_test.go
+++ b/integration/integration_test.go
@@ -4586,10 +4586,9 @@ func testControlMaster(t *testing.T, suite *integrationTestSuite) {
 			inRecordLocation: types.RecordAtNode,
 		},
 		// Run tests when Teleport is recording sessions at the proxy
-		// (temporarily disabled, see https://github.com/gravitational/teleport/issues/16224)
-		// {
-		// 	inRecordLocation: types.RecordAtProxy,
-		// },
+		{
+			inRecordLocation: types.RecordAtProxy,
+		},
 	}
 
 	for _, tt := range tests {
@@ -4631,10 +4630,10 @@ func testControlMaster(t *testing.T, suite *integrationTestSuite) {
 			require.NoError(t, err)
 			defer helpers.CloseAgent(teleAgent, socketDirPath)
 
-			// Create and run an exec command twice with the passed in ControlPath. This
-			// will cause re-use of the connection and creation of two sessions within
-			// the connection.
-			for range 2 {
+			testConnection := func(t *testing.T) {
+				// Create and run an exec command twice with the passed in ControlPath. This
+				// will cause re-use of the connection and creation of two sessions within
+				// the connection.
 				execCmd, err := helpers.ExternalSSHCommand(helpers.CommandOptions{
 					ForcePTY:     true,
 					ForwardAgent: true,
@@ -4658,6 +4657,13 @@ func testControlMaster(t *testing.T, suite *integrationTestSuite) {
 				require.NoError(t, err, "stderr=%q", stderr)
 				require.True(t, strings.HasSuffix(strings.TrimSpace(string(output)), "hello"))
 			}
+
+			t.Run("first connection", func(t *testing.T) {
+				testConnection(t)
+			})
+			t.Run("second connection", func(t *testing.T) {
+				testConnection(t)
+			})
 		})
 	}
 }

